sql/core/src/test/scala/org/apache/spark/sql/sources/TableScanSuite.scala

```diff
@@ -109,6 +110,19 @@ case class AllDataTypesScan(
   }
 }
 
+class LegacyTimestampSource extends RelationProvider {
```

Comment:
can we use the data source write API as an example? For read API, another fix we should do later is to let Spark accept both old and new datetime classes, which will make this test useless.

Reply:
I am not quite sure that I got your point. You proposed to add a config for `collect()` (which can control ONLY decoding, obviously) and to write a test using the DS write API? How (does `collect()` involve the write API)? Could you clarify, please.